src/Features/CSharp/Portable/UpgradeProject/CSharpUpgradeProjectCodeFixProvider.cs

```diff
@@ -0,0 +1,91 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Linq;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.UpgradeProject;
+
+namespace Microsoft.CodeAnalysis.CSharp.UpgradeProject
+{
+    [ExportCodeFixProvider(LanguageNames.CSharp), Shared]
+    internal class CSharpUpgradeProjectCodeFixProvider : AbstractUpgradeProjectCodeFixProvider
+    {
+        private const string CS8022 = nameof(CS8022); // error CS8022: Feature is not available in C# 1.  Please use language version X or greater.
+        private const string CS8023 = nameof(CS8023); // error CS8023: Feature is not available in C# 2.  Please use language version X or greater.
+        private const string CS8024 = nameof(CS8024); // error CS8024: Feature is not available in C# 3.  Please use language version X or greater.
+        private const string CS8025 = nameof(CS8025); // error CS8025: Feature is not available in C# 4.  Please use language version X or greater.
+        private const string CS8026 = nameof(CS8026); // error CS8026: Feature is not available in C# 5.  Please use language version X or greater.
+        private const string CS8059 = nameof(CS8059); // error CS8059: Feature is not available in C# 6.  Please use language version X or greater.
+
+        private static readonly ImmutableArray<string> s_diagnostics =
+            ImmutableArray.Create(CS8022, CS8023, CS8024, CS8025, CS8026, CS8059);
+        public override ImmutableArray<string> FixableDiagnosticIds { get; } = s_diagnostics;
+
+        public override string UpgradeThisProjectResource => CSharpFeaturesResources.Upgrade_this_project_to_csharp_language_version_0;
+        public override string UpgradeAllProjectsResource => CSharpFeaturesResources.Upgrade_all_projects_to_csharp_language_version_0;
+
+        public override IEnumerable<string> SuggestedVersions(ImmutableArray<Diagnostic> diagnostics)
+        {
+            var required = RequiredVersion(diagnostics);
+
+            var builder = ArrayBuilder<string>.GetInstance(1);
+
+            var generic = required <= LanguageVersion.Default.MapSpecifiedToEffectiveVersion()
```

Comment:
Translating this to English: "we always upgrade them to the default, even if there's a lower version that would still satisfy it?"